app/Filament/Resources/AchievementResource/RelationManagers/AuthorshipCreditsRelationManager.php

```diff
@@ -58,6 +58,24 @@ public function table(Table $table): Table
                         $achievement = $this->ownerRecord;
 
                         $task = $data['task'];
+                        $userId = (int) $data['user_id'];
+
+                        // First, check for and restore any soft-deleted record.
+                        $existingRecord = AchievementAuthor::withTrashed()
+                            ->whereAchievementId($achievement->id)
+                            ->whereUserId($userId)
+                            ->whereTask($task)
+                            ->first();
+
+                        if ($existingRecord) {
+                            if ($existingRecord->trashed()) {
+                                $existingRecord->restore();
+
+                                return $existingRecord;
```

Comment:
this return is redundant with the return on line 77.

Reply:
Removed in latest.